docs/reference/schemas/definitions/resourceKind.md

```diff
@@ -0,0 +1,358 @@
+            output:  >-
+              [reference(
+                resourceId('Microsoft/OSInfo', 'Deeply nested OSInfo')
+              ).actualState]
+          dependsOn:
+            - "[resourceId('Microsoft/OSInfo', 'Deeply nested OSInfo')]"
+        - name: Deeply nested OSInfo
+          type: Microsoft.OSInfo
+          properties: {}
+```
+
+In every case, the references and dependencies are to adjacent instances in the configuration.
+Instances at the top level only depend on or reference other instances at the top level. Instances
+nested in the top-level group only depend on or reference other nested instances in the same group.
+The deeply nested instances defined in the nested group only depend on or reference other deeply
+nested instances in the same group.
+
+Putting the configuration together, you get this full document:
+
+```yaml
+# yaml-language-server: $schema=https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/bundled/resource/manifest.vscode.json
+$schema: https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/config/document.json
+resources:
+# The top level echo references and depends on the top-level OSInfo.
+- name: Top level echo
+  type: Test/Echo
+  properties:
+    output:  >-
+      [reference(
+        resourceId('Microsoft/OSInfo', 'Top level OSInfo')
+      ).actualState]
+  dependsOn:
+    - "[resourceId('Microsoft/OSInfo', 'Top level OSInfo')]"
+# The top level OSInfo depends on the top-level Group.
+- name: Top level OSInfo
+  type: Microsoft/OSInfo
+  properties: {}
+  dependsOn:
+    - "[resourceId('Microsoft.DSC/Group', 'Top level group')]"
+- name: Top level group
+  type: Microsoft.DSC/Group
+  properties:
+    $schema: https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/config/document.json
+    resources:
+    # The nested echo references and depends on the adjacent nested OSInfo.
+    - name: Nested echo
+      type: Test/Echo
+      properties:
+        output:  >-
+          [reference(
+            resourceId('Microsoft/OSInfo', 'Nested OSInfo')
+          ).actualState]
+      dependsOn:
+        - "[resourceId('Microsoft/OSInfo', 'Nested OSInfo')]"
+    # The nested OSInfo depends on the adjacent nested Group.
+    - name: Nested OSInfo
+      type: Microsoft/OSInfo
+      properties: {}
+    - name: Nested Group
+      type: Microsoft.DSC/Group
+      properties:
+        $schema: https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/config/document.json
+        resources:
+        # The deeply nested echo references and depends on the adjacent
+        # deeply nested OSInfo.
+        - name: Deeply nested echo
+          type: Test/Echo
+          properties:
+            output:  >-
+              [reference(
+                resourceId('Microsoft/OSInfo', 'Deeply nested OSInfo')
+              ).actualState]
+          dependsOn:
+            - "[resourceId('Microsoft/OSInfo', 'Deeply nested OSInfo')]"
```

Comment:
would it be useful to have an example of depending on more than one resource?  LIke maybe a group and an individual resource?